pkg/cli/image/manifest/manifest.go

```diff
@@ -276,47 +276,78 @@ func FirstManifest(ctx context.Context, from imagereference.DockerImageReference
 }
 
 // ManifestToImageConfig takes an image manifest and converts it into a structured object.
-func ManifestToImageConfig(ctx context.Context, srcManifest distribution.Manifest, blobs distribution.BlobService, location ManifestLocation) (*dockerv1client.DockerImageConfig, []distribution.Descriptor, error) {
+func ManifestToImageConfig(ctx context.Context, srcManifest distribution.Manifest, blobs distribution.BlobService, manifests distribution.ManifestService, location ManifestLocation, platformSpec string) (*dockerv1client.DockerImageConfig, []distribution.Descriptor, string, error) {
+	switch t := srcManifest.(type) {
+	case *manifestlist.DeserializedManifestList:
+		// In case of 'oc adm release new' with a component image substituted and that component image is multi-arch,
```

Comment:
This comment is wierdly specific to a oc, you really need to describe in the godoc how ManifestToImageConfig handles manifest lists.  And passing platformSpec in here is kind of wierd - platform spec is a high level concern of the `oc` command, not of low level code like this utility method.  You probably need to provide something like a handler function that, if specified, is passed the manifest list and tells you which image to select (something like `listSelectionFn func(DeserializedManifestList) (Manifest, error)` which returns a manifest, nil, or an error).  

Also, you should only fetch the child manifest AFTER you filter, not before (this will load O(N) manifests, instead of O(1) which it should be).

Reply:
restructured, updated the godoc, no need to return the platformSpec there as it is the DockerImageConfig.OS/Architecture already returned.  

I'm not sure what you mean w/ fetching the child manifest after filtering. The filtering [here](https://github.com/openshift/oc/blob/master/pkg/cli/image/extract/extract.go#L366) [to filter operators](https://github.com/openshift/oc/blob/master/pkg/cli/admin/release/new.go#L978)?

Reply:
updated